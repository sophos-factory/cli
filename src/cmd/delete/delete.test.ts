/* eslint-disable @typescript-eslint/no-non-null-assertion */

import { execute } from '../../../tests/helpers/execute';
import { loadFixtures } from '../../../tests/fixtures';

beforeAll(async () => {
  return await loadFixtures();
});

describe('factoryctl delete', () => {
  test('throws on missing subcommand', async () => {
    await expect(
      execute(['delete'], { token: process.env.FACTORY_DYNAMIC_AUTH_TOKEN! })
    ).rejects.toMatchSnapshot();
  });

  describe('credential', () => {
    test('throws on missing arguments', async () => {
      await expect(
        execute(['delete', 'credential'], {
          token: process.env.FACTORY_DYNAMIC_AUTH_TOKEN!
        })
      ).rejects.toMatchSnapshot();
    });

    test('throws on missing --project-id option', async () => {
      await expect(
        execute(['delete', 'credential', 'credential-id'], {
          token: process.env.FACTORY_DYNAMIC_AUTH_TOKEN!
        })
      ).rejects.toMatchSnapshot();
    });
  });

  describe('job', () => {
    test('throws on missing arguments', async () => {
      await expect(
        execute(['delete', 'job'], {
          token: process.env.FACTORY_DYNAMIC_AUTH_TOKEN!
        })
      ).rejects.toMatchSnapshot();
    });

    test('throws on missing --project-id option', async () => {
      await expect(
        execute(['delete', 'job', 'job-id'], {
          token: process.env.FACTORY_DYNAMIC_AUTH_TOKEN!
        })
      ).rejects.toMatchSnapshot();
    });
  });

  describe('pipeline', () => {
    test('throws on missing arguments', async () => {
      await expect(
        execute(['delete', 'pipeline'], {
          token: process.env.FACTORY_DYNAMIC_AUTH_TOKEN!
        })
      ).rejects.toMatchSnapshot();
    });
  });

  describe('project', () => {
    test('throws on missing arguments', async () => {
      await expect(
        execute(['delete', 'project'], {
          token: process.env.FACTORY_DYNAMIC_AUTH_TOKEN!
        })
      ).rejects.toMatchSnapshot();
    });
  });

  describe('runner', () => {
    test('throws on missing arguments', async () => {
      await expect(
        execute(['delete', 'runner'], {
          token: process.env.FACTORY_DYNAMIC_AUTH_TOKEN!
        })
      ).rejects.toMatchSnapshot();
    });
    test('throws on missing organization-id option', async () => {
      await expect(
        execute(['delete', 'runner', 'runner-id'], {
          token: process.env.FACTORY_DYNAMIC_AUTH_TOKEN!
        })
      ).rejects.toMatchSnapshot();
    });
  });

  describe('runner pool', () => {
    test('throws on missing arguments', async () => {
      await expect(
        execute(['delete', 'runner-pool'], {
          token: process.env.FACTORY_DYNAMIC_AUTH_TOKEN!
        })
      ).rejects.toMatchSnapshot();
    });

    test('throws on missing organization-id option', async () => {
      await expect(
        execute(['delete', 'runner-pool', 'runner-pool-id'], {
          token: process.env.FACTORY_DYNAMIC_AUTH_TOKEN!
        })
      ).rejects.toMatchSnapshot();
    });
  });
});
