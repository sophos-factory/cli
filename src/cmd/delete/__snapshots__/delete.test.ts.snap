// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`factoryctl delete credential throws on missing --project-id option 1`] = `
"factoryctl delete credential [credential-id]

Delete a credential

Positionals:
  credential-id  ID of the credential to deleted  [string] [default: read from stdin]

Options:
      --version       Show version number  [boolean]
      --address       Address of the Factory API server. Required for all commands except "organization(s)  [string] [default: FACTORY_ADDRESS if set, otherwise https://api.dev.factory.sophos.com/v1]
      --auth-address  Address of the Factory Auth API server. Required for "organization(s)" commands  [string] [default: FACTORY_AUTH_ADDRESS if set, otherwise https://auth.dev.factory.sophos.com/v1]
      --auth-token    Authentication token  [string] [default: FACTORY_AUTH_TOKEN environment variable]
      --filter        Filter output using JSONPath  [string]
      --format        Output format  [choices: "wide", "json", "yaml"] [default: "wide"]
  -v, --verbose       Print detailed output  [boolean]
      --help          Show help  [boolean]
      --project-id    ID of the project containing the credential  [string] [required]

Missing required argument: project-id
"
`;

exports[`factoryctl delete credential throws on missing arguments 1`] = `
"factoryctl delete credential [credential-id]

Delete a credential

Positionals:
  credential-id  ID of the credential to deleted  [string] [default: read from stdin]

Options:
      --version       Show version number  [boolean]
      --address       Address of the Factory API server. Required for all commands except "organization(s)  [string] [default: FACTORY_ADDRESS if set, otherwise https://api.dev.factory.sophos.com/v1]
      --auth-address  Address of the Factory Auth API server. Required for "organization(s)" commands  [string] [default: FACTORY_AUTH_ADDRESS if set, otherwise https://auth.dev.factory.sophos.com/v1]
      --auth-token    Authentication token  [string] [default: FACTORY_AUTH_TOKEN environment variable]
      --filter        Filter output using JSONPath  [string]
      --format        Output format  [choices: "wide", "json", "yaml"] [default: "wide"]
  -v, --verbose       Print detailed output  [boolean]
      --help          Show help  [boolean]
      --project-id    ID of the project containing the credential  [string] [required]

Missing required argument: project-id
"
`;

exports[`factoryctl delete job throws on missing --project-id option 1`] = `
"factoryctl delete job [job-id]

Delete a job

Positionals:
  job  ID of the job to delete  [string]

Options:
      --version       Show version number  [boolean]
      --address       Address of the Factory API server. Required for all commands except "organization(s)  [string] [default: FACTORY_ADDRESS if set, otherwise https://api.dev.factory.sophos.com/v1]
      --auth-address  Address of the Factory Auth API server. Required for "organization(s)" commands  [string] [default: FACTORY_AUTH_ADDRESS if set, otherwise https://auth.dev.factory.sophos.com/v1]
      --auth-token    Authentication token  [string] [default: FACTORY_AUTH_TOKEN environment variable]
      --filter        Filter output using JSONPath  [string]
      --format        Output format  [choices: "wide", "json", "yaml"] [default: "wide"]
  -v, --verbose       Print detailed output  [boolean]
      --help          Show help  [boolean]
      --project-id    ID of the project containing the job  [string] [required]
      --job-id  [default: read from stdin]

Missing required argument: project-id
"
`;

exports[`factoryctl delete job throws on missing arguments 1`] = `
"factoryctl delete job [job-id]

Delete a job

Positionals:
  job  ID of the job to delete  [string]

Options:
      --version       Show version number  [boolean]
      --address       Address of the Factory API server. Required for all commands except "organization(s)  [string] [default: FACTORY_ADDRESS if set, otherwise https://api.dev.factory.sophos.com/v1]
      --auth-address  Address of the Factory Auth API server. Required for "organization(s)" commands  [string] [default: FACTORY_AUTH_ADDRESS if set, otherwise https://auth.dev.factory.sophos.com/v1]
      --auth-token    Authentication token  [string] [default: FACTORY_AUTH_TOKEN environment variable]
      --filter        Filter output using JSONPath  [string]
      --format        Output format  [choices: "wide", "json", "yaml"] [default: "wide"]
  -v, --verbose       Print detailed output  [boolean]
      --help          Show help  [boolean]
      --project-id    ID of the project containing the job  [string] [required]
      --job-id  [default: read from stdin]

Missing required argument: project-id
"
`;

exports[`factoryctl delete pipeline throws on missing arguments 1`] = `
"factoryctl delete pipeline [pipeline-id]

Delete a pipeline

Positionals:
  pipeline-id  ID of the pipeline to delete  [string] [default: read from stdin]

Options:
      --version       Show version number  [boolean]
      --address       Address of the Factory API server. Required for all commands except "organization(s)  [string] [default: FACTORY_ADDRESS if set, otherwise https://api.dev.factory.sophos.com/v1]
      --auth-address  Address of the Factory Auth API server. Required for "organization(s)" commands  [string] [default: FACTORY_AUTH_ADDRESS if set, otherwise https://auth.dev.factory.sophos.com/v1]
      --auth-token    Authentication token  [string] [default: FACTORY_AUTH_TOKEN environment variable]
      --filter        Filter output using JSONPath  [string]
      --format        Output format  [choices: "wide", "json", "yaml"] [default: "wide"]
  -v, --verbose       Print detailed output  [boolean]
      --help          Show help  [boolean]
      --project-id    ID of the project containing the pipeline  [string] [required]

Missing required argument: project-id
"
`;

exports[`factoryctl delete project throws on missing arguments 1`] = `
"factoryctl delete project [project-id]

Delete a project

Positionals:
  project-id  ID of the project to delete  [string] [default: read from stdin]

Options:
      --version       Show version number  [boolean]
      --address       Address of the Factory API server. Required for all commands except "organization(s)  [string] [default: FACTORY_ADDRESS if set, otherwise https://api.dev.factory.sophos.com/v1]
      --auth-address  Address of the Factory Auth API server. Required for "organization(s)" commands  [string] [default: FACTORY_AUTH_ADDRESS if set, otherwise https://auth.dev.factory.sophos.com/v1]
      --auth-token    Authentication token  [string] [default: FACTORY_AUTH_TOKEN environment variable]
      --filter        Filter output using JSONPath  [string]
      --format        Output format  [choices: "wide", "json", "yaml"] [default: "wide"]
  -v, --verbose       Print detailed output  [boolean]
      --help          Show help  [boolean]

Project ID must be provided
"
`;

exports[`factoryctl delete runner pool throws on missing arguments 1`] = `
"factoryctl delete runner-pool [runner-pool-id]

Delete a runner pool

Positionals:
  runner-pool-id  ID of the runner pool to delete  [string] [default: read from stdin]

Options:
      --version          Show version number  [boolean]
      --address          Address of the Factory API server. Required for all commands except "organization(s)  [string] [default: FACTORY_ADDRESS if set, otherwise https://api.dev.factory.sophos.com/v1]
      --auth-address     Address of the Factory Auth API server. Required for "organization(s)" commands  [string] [default: FACTORY_AUTH_ADDRESS if set, otherwise https://auth.dev.factory.sophos.com/v1]
      --auth-token       Authentication token  [string] [default: FACTORY_AUTH_TOKEN environment variable]
      --filter           Filter output using JSONPath  [string]
      --format           Output format  [choices: "wide", "json", "yaml"] [default: "wide"]
  -v, --verbose          Print detailed output  [boolean]
      --help             Show help  [boolean]
      --organization-id  ID of the organization containing the runner pool  [string] [required]

Missing required argument: organization-id
"
`;

exports[`factoryctl delete runner pool throws on missing organization-id option 1`] = `
"factoryctl delete runner-pool [runner-pool-id]

Delete a runner pool

Positionals:
  runner-pool-id  ID of the runner pool to delete  [string] [default: read from stdin]

Options:
      --version          Show version number  [boolean]
      --address          Address of the Factory API server. Required for all commands except "organization(s)  [string] [default: FACTORY_ADDRESS if set, otherwise https://api.dev.factory.sophos.com/v1]
      --auth-address     Address of the Factory Auth API server. Required for "organization(s)" commands  [string] [default: FACTORY_AUTH_ADDRESS if set, otherwise https://auth.dev.factory.sophos.com/v1]
      --auth-token       Authentication token  [string] [default: FACTORY_AUTH_TOKEN environment variable]
      --filter           Filter output using JSONPath  [string]
      --format           Output format  [choices: "wide", "json", "yaml"] [default: "wide"]
  -v, --verbose          Print detailed output  [boolean]
      --help             Show help  [boolean]
      --organization-id  ID of the organization containing the runner pool  [string] [required]

Missing required argument: organization-id
"
`;

exports[`factoryctl delete runner throws on missing arguments 1`] = `
"factoryctl delete runner [runner-id]

Delete a runner

Positionals:
  runner-id  ID of the runner to delete  [string] [default: read from stdin]

Options:
      --version          Show version number  [boolean]
      --address          Address of the Factory API server. Required for all commands except "organization(s)  [string] [default: FACTORY_ADDRESS if set, otherwise https://api.dev.factory.sophos.com/v1]
      --auth-address     Address of the Factory Auth API server. Required for "organization(s)" commands  [string] [default: FACTORY_AUTH_ADDRESS if set, otherwise https://auth.dev.factory.sophos.com/v1]
      --auth-token       Authentication token  [string] [default: FACTORY_AUTH_TOKEN environment variable]
      --filter           Filter output using JSONPath  [string]
      --format           Output format  [choices: "wide", "json", "yaml"] [default: "wide"]
  -v, --verbose          Print detailed output  [boolean]
      --help             Show help  [boolean]
      --organization-id  ID of the organization containing the runner  [string] [required]

Missing required argument: organization-id
"
`;

exports[`factoryctl delete runner throws on missing organization-id option 1`] = `
"factoryctl delete runner [runner-id]

Delete a runner

Positionals:
  runner-id  ID of the runner to delete  [string] [default: read from stdin]

Options:
      --version          Show version number  [boolean]
      --address          Address of the Factory API server. Required for all commands except "organization(s)  [string] [default: FACTORY_ADDRESS if set, otherwise https://api.dev.factory.sophos.com/v1]
      --auth-address     Address of the Factory Auth API server. Required for "organization(s)" commands  [string] [default: FACTORY_AUTH_ADDRESS if set, otherwise https://auth.dev.factory.sophos.com/v1]
      --auth-token       Authentication token  [string] [default: FACTORY_AUTH_TOKEN environment variable]
      --filter           Filter output using JSONPath  [string]
      --format           Output format  [choices: "wide", "json", "yaml"] [default: "wide"]
  -v, --verbose          Print detailed output  [boolean]
      --help             Show help  [boolean]
      --organization-id  ID of the organization containing the runner  [string] [required]

Missing required argument: organization-id
"
`;

exports[`factoryctl delete throws on missing subcommand 1`] = `
"factoryctl delete

Delete specified resource

Commands:
  factoryctl delete credential [credential-id]    Delete a credential
  factoryctl delete job [job-id]                  Delete a job
  factoryctl delete pipeline [pipeline-id]        Delete a pipeline
  factoryctl delete project [project-id]          Delete a project
  factoryctl delete runner [runner-id]            Delete a runner
  factoryctl delete runner-pool [runner-pool-id]  Delete a runner pool

Options:
      --version       Show version number  [boolean]
      --address       Address of the Factory API server. Required for all commands except "organization(s)  [string] [default: FACTORY_ADDRESS if set, otherwise https://api.dev.factory.sophos.com/v1]
      --auth-address  Address of the Factory Auth API server. Required for "organization(s)" commands  [string] [default: FACTORY_AUTH_ADDRESS if set, otherwise https://auth.dev.factory.sophos.com/v1]
      --auth-token    Authentication token  [string] [default: FACTORY_AUTH_TOKEN environment variable]
      --filter        Filter output using JSONPath  [string]
      --format        Output format  [choices: "wide", "json", "yaml"] [default: "wide"]
  -v, --verbose       Print detailed output  [boolean]
      --help          Show help  [boolean]

Command must be specified.
"
`;
